### PR TITLE
Add golangci-lint checks

### PR DESCRIPTION
**Pull request by Fedosin**
This commit adds lint validations from golangci-lint project. Also it introduces "verify" target, that combines "fmt", "vet" and "lint".

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,8 +71,13 @@ CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
 controller-gen:
 	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
 
+# Download controller-gen locally if necessary
+GOLINT = $(shell pwd)/bin/golint
```
Comment (JoelSpeed) [inline Makefile:75]: Why does this comment have `controller-gen` in it? 🤔

File: pkg/dependencymagnet/dependencymagnet.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// +build tools
```
Comment (JoelSpeed) [inline pkg/dependencymagnet/dependencymagnet.go:1]: Is this where we want to keep this? I've seen in a lot of places to just have a `tools.go` in the root of the project, WDYT?

**Review by JoelSpeed — COMMENTED** [review]
At the moment it seems to be linting the vendor directory too, can we make sure that's not the case?

**Comment by JoelSpeed** [comment]
Yeah this looks good now, thanks

I think at some point we should review the linters we have enabled and create a full config as a team with standards that we want to uphold for all of our repos

/approve

**Comment by elmiko** [comment]
talked with @Fedosin in slack, overriding the failing tests since they are not related to this change.
/override ci/prow/e2e-aws-upgrade
/override ci/prow/e2e-aws-ccm-install

**Pull request merged**